### PR TITLE
Don't import dev cert

### PR DESCRIPTION
**Pull request by JoshLove-msft**
Instead use the ASP.NET env vars to just point to the cert location.

### CODE REVIEW

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json [hunk 1]
```diff
@@ -29,7 +29,7 @@
     }
   ],
   "Variables": {
-    "DateTimeOffsetNow": "2021-03-05T16:18:36.8937492-08:00",
+    "DateTimeOffsetNow": "2021-03-06T00:18:36.8937492Z",
```
Comment (JoshLove-msft) [inline sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json:32]: This wasn't updated in the Autorest PR - I just happen to use Event Grid for testing Test Proxy.

File: sdk/core/Azure.Core.TestFramework/src/TestProxy.cs [hunk 2]
```diff
@@ -71,16 +71,25 @@ private TestProxy(string proxyPath)
...
+                        "dotnet-devcert.pfx"),
+                    ["ASPNETCORE_Kestrel__Certificates__Default__Password"] = "password"
                 }
             };
 
-            if (!TestEnvironment.GlobalIsRunningInCI)
+            _testProxyProcess = Process.Start(testProxyProcessInfo);
+
+            // if the process immediately exits, surface the error
+            if (_testProxyProcess.HasExited)
```
Comment (pakrym) [inline sdk/core/Azure.Core.TestFramework/src/TestProxy.cs:88]: this is very racy. Does it ever get hit?

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json (hunk 1 above)
Comment (pakrym) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json:32]: Why is this passing?

File: sdk/core/Azure.Core.TestFramework/src/TestProxy.cs [hunk 3]
```diff
@@ -114,7 +123,9 @@ private TestProxy(string proxyPath)
 
             if (_proxyPortHttp == null || _proxyPortHttps == null)
             {
-                throw new InvalidOperationException("Failed to start the test proxy.");
+                throw new InvalidOperationException("Failed to start the test proxy. One or both of the ports was not populated." + Environment.NewLine +
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.TestFramework/src/TestProxy.cs:126]: Thank you for improving the warning message :)

File: sdk/core/Azure.Core.TestFramework/src/TestProxy.cs [hunk 4]
```diff
@@ -71,16 +71,25 @@ private TestProxy(string proxyPath)
                 EnvironmentVariables =
                 {
                     ["ASPNETCORE_URLS"] = $"http://{IpAddress}:0;https://{IpAddress}:0",
-                    ["Logging__LogLevel__Microsoft"] = "Information"
+                    ["Logging__LogLevel__Microsoft.Hosting.Lifetime"] = "Information",
+                    ["ASPNETCORE_Kestrel__Certificates__Default__Path"] = Path.Combine(
```
Comment (annelo-msft) [inline sdk/core/Azure.Core.TestFramework/src/TestProxy.cs:75]: Will this handle the case that the cert isn't here and/or prompt you to install it?

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json:32]: It's not passing locally - not sure how CI passed.

File: sdk/core/Azure.Core.TestFramework/src/TestProxy.cs (hunk 4 above)
Comment (JoshLove-msft) [reply sdk/core/Azure.Core.TestFramework/src/TestProxy.cs:75]: With this change you won't have to install the cert anymore as we just specify the location of the cert that is in the repo.

File: sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json (hunk 1 above)
Comment (pakrym) [reply sdk/eventgrid/Azure.Messaging.EventGrid/tests/SessionRecords/EventGridClientLiveTests/CanPublishSingleEventAsync.json:32]: We need to investigate.

**Review by pakrym — COMMENTED** [review]
Revert the recording change.

**Review by pakrym — APPROVED** [review]

**Pull request merged**